pageserver/src/pgdatadir_mapping.rs

```diff
@@ -1464,6 +1470,10 @@ impl<'a> DatadirModification<'a> {
         blknum: BlockNumber,
         rec: NeonWalRecord,
     ) -> anyhow::Result<()> {
+        if !self.tline.tenant_shard_id.is_shard_zero() {
```

Comment:
Why aren't these (debug) assertions?

Reply:
Yeah, it should be an assertion.  Changed in bf07db61c5783c518dbb673773dafb35f460042f